### PR TITLE
test: remove avalanche tests

### LINKED ISSUE #772: Use network specific config for tests with permit2

### PR DESCRIPTION
**Pull request by MathisGD**
As they aren't maintained, and they are quite useless

Also fixes #772

### CODE REVIEW

File: script/EthEModeDeploy.s.sol
Comment (QGarchery) [inline script/EthEModeDeploy.s.sol:1]: We can change the makefile as well, in particular the rule `deploy-emode-%` that doesn't depend on the network anymore

File: README.md
Comment (QGarchery) [inline README.md:1]: We can remove many things related to the network. For example the `network` input in `forge-test.yml`, as well as most of the occurrences of the `ethereum-mainnet` keyword in this repository

**Comment by MathisGD** [comment]
Tests will pass once merged on main

**Review by QGarchery — DISMISSED** [review]

File: test/integration/TestIntegrationPermit2.sol [hunk 1]
```diff
@@ -72,7 +72,7 @@ contract TestIntegrationPermit2 is IntegrationTest {
         morpho.supplyWithPermit(market.underlying, amount, delegator, DEFAULT_MAX_ITERATIONS, deadline, sig);
         /// The maximum gap needs to be 4 because sometimes the timestamp is very big, otherwise the test reverts.
         assertApproxEqAbs(
-            morpho.supplyBalance(market.underlying, delegator), balanceSupplyBefore + amount, 4, "Incorrect Supply"
+            morpho.supplyBalance(market.underlying, delegator), balanceSupplyBefore + amount, 7, "Incorrect Supply"
```
Comment (QGarchery) [inline test/integration/TestIntegrationPermit2.sol:75]: The above comment mentions 4 as the bound, maybe just remove the comment ?

**Review by QGarchery — APPROVED** [review]

File: test/integration/TestIntegrationPermit2.sol [hunk 2]
```diff
@@ -72,7 +72,7 @@ contract TestIntegrationPermit2 is IntegrationTest {
         morpho.supplyWithPermit(market.underlying, amount, delegator, DEFAULT_MAX_ITERATIONS, deadline, sig);
         /// The maximum gap needs to be 4 because sometimes the timestamp is very big, otherwise the test reverts.
```
Comment (MathisGD) [inline test/integration/TestIntegrationPermit2.sol:73]:
```suggestion
```

**Review by Rubilmax — APPROVED** [review]

**Pull request merged**